src/test/scala/ch/epfl/bluebrain/nexus/admin/directives/QueryDirectivesSpec.scala

```diff
@@ -0,0 +1,150 @@
+package ch.epfl.bluebrain.nexus.admin.directives
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.Directives.{complete, get, handleExceptions, handleRejections}
+import akka.http.scaladsl.server.Route
+import akka.http.scaladsl.testkit.ScalatestRouteTest
+import ch.epfl.bluebrain.nexus.admin.CommonRejection.IllegalParameter
+import ch.epfl.bluebrain.nexus.admin.Error
+import ch.epfl.bluebrain.nexus.admin.Error._
+import ch.epfl.bluebrain.nexus.admin.config.AppConfig
+import ch.epfl.bluebrain.nexus.admin.config.Contexts.errorCtxUri
+import ch.epfl.bluebrain.nexus.admin.marshallers.instances._
+import ch.epfl.bluebrain.nexus.admin.routes.{ExceptionHandling, RejectionHandling}
+import ch.epfl.bluebrain.nexus.commons.types.search.Pagination
+import ch.epfl.bluebrain.nexus.rdf.Iri.Path
+import org.mockito.integrations.scalatest.IdiomaticMockitoFixture
+import org.scalatest.{EitherValues, Matchers, WordSpecLike}
+import org.scalatest.concurrent.ScalaFutures
+
+class QueryDirectivesSpec
+    extends WordSpecLike
+    with ScalatestRouteTest
+    with Matchers
+    with ScalaFutures
+    with EitherValues
+    with IdiomaticMockitoFixture {
+
+  private def routes(inner: Route): Route =
+    handleExceptions(ExceptionHandling.handler) {
+      handleRejections(RejectionHandling.handler) {
+        inner
+      }
+    }
+
+  "Query directives" should {
+    "handle pagination" in {
+      def paginated(from: Long, size: Int) = (get & QueryDirectives.paginated(AppConfig.PaginationConfig(0, 50, 100))) {
+        pagination =>
+          pagination shouldEqual Pagination(from, size)
+          complete(StatusCodes.Accepted)
+      }
+
+      Get("/") ~> routes(paginated(0L, 50)) ~> check {
+        status shouldEqual StatusCodes.Accepted
+      }
+      Get("/?size=42") ~> routes(paginated(0L, 42)) ~> check {
+        status shouldEqual StatusCodes.Accepted
+      }
+      Get("/?from=1") ~> routes(paginated(1L, 50)) ~> check {
+        status shouldEqual StatusCodes.Accepted
+      }
+      Get("/?from=1&size=42") ~> routes(paginated(1L, 42)) ~> check {
+        status shouldEqual StatusCodes.Accepted
+      }
+      Get("/?from=1&size=-42") ~> routes(paginated(1L, 0)) ~> check {
```

Comment:
shouldn't this be a `BadRequest`?

Reply:
We handle it like that in other services.